openedx/openedx-platform
# feat: add share link button when hide from toc is enabled in sections

## Description
This PR adds a new button in the child subsections of sections with Hide From TOC enabled. This button displays a new modal with two tabs. The first tab displays a button that allows you to copy the link of that subsection to the clipboard. The second tab displays a button that allows you to copy the embedded link of the same subsection to the clipboard.

In this way, instructors will be able to pass the link to their students so that they can access the subsections that are hidden from the interface.

## Supporting information

These changes are part of the effort made to implement [Feature Enhancement Proposal: Hide Sections from Course Outline](https://openedx.atlassian.net/wiki/spaces/OEPM/pages/3853975595/Feature+Enhancement+Proposal+Hide+Sections+from+course+outline)

## Dependencies
> [!IMPORTANT]
> This PR needs the changes made from [this PR](https://github.com/openedx/edx-platform/pull/33952), which includes the new section visibility option from Studio UI.

## Testing instructions

1. In your environment add a mount of `edx-platform` with `tutor mounts add <path/to/edx-platform>` 
2. Move into this branch with `git checkout bav/add-share-link-button`
3. Get the changes from [this PR](https://github.com/openedx/edx-platform/pull/33952) with `git rebase MJG/configurable-hide-from-toc` and resolves conflicts by accepting both changes.
4. Add this setting in your environment: `FEATURES["ENABLE_HIDE_FROM_TOC_UI"] = True`
5. Run `tutor dev exec lms openedx-assets build --env=dev` to regenerate the assets.
6. Create your course with sections and subsections.
7.  In the section settings (⚙️) go to the **Visibility** tab > ☑️ **Hide in Course Outline, accessible via Link**
8. Now, you should see a button to share the link in the child subsections of that section.
9. Press the button and see the new modal.
10. Navigate between the tabs and copy the subsection link, either full page or embedded.

https://github.com/openedx/edx-platform/assets/64033729/37611202-fefc-41db-ae78-ad15b95e556c

## Deadline
This effort is part of the Spanish consortium project, so it'd be ideal to merge this before the end of the project.
